GridSplitter - Fixed issue where the grid would overflow when resizing columns with minimum width

kbrons opened the pull request:
Issue: #1894 

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one ore more that apply to this PR -->

- Bugfix
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
When resizing columns on a grid with one or more automatic width columns with minimum width, the grid starts getting bigger and ultimately overflows.

![bug](https://user-images.githubusercontent.com/14252330/39260908-31ffd968-4891-11e8-8807-79c406bf326a.gif)

## What is the new behavior?
The resizing stops whenever the minimum width of any automatic width column would be violated.

![fixed](https://user-images.githubusercontent.com/14252330/39260915-3613e7b0-4891-11e8-8fcd-661fe4876f5c.gif)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/Microsoft/UWPCommunityToolkit/blob/master/docs/.template.md). (for bug fixes / features)
- [ ] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/Microsoft/UWPCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [X] Contains **NO** breaking changes

skendrot commented:
If the bottom gif is the new behavior it's still wrong. The Grid should not expand on the right if you're dragging from the left. Likewise if dragging from the right it shouldn't move the right "wall"

kbrons commented:
This fix is for a specific case having two automatic width columns (left and right) and a fixed width column (center) that provoked an overflow of the grid. Investigating some more, the behavior you are describing happens when all the columns have automatic width (gif below).

![all automatic scenario](https://user-images.githubusercontent.com/14252330/39266422-5e218dfc-48a0-11e8-902f-8c3cce954547.gif)

The thing is, that behavior wasn't correctly defined or implemented for the case of this fix, nor the case where all the columns have a fixed width (gif below).

![all fixed scenario](https://user-images.githubusercontent.com/14252330/39266605-d920853a-48a0-11e8-9048-52914f75b365.gif)

Taking that into account, I think the behavior of the component should be redefined and reworked, along with all the configurations of the control tested to make sure they follow that behavior.
I think that the best would be to create a new issue to redefine this component's behavior and split the efforts.

kbrons commented:
I've been taking a look at the code and testing some stuff, and found that leaving only this part of the column resizing logic, everything seems to be working at first glance. Does anyone know why the code for resizing when the current or sibling columns have fixed width was added? 

This is the logic that I was testing out (along with checking that neither the current and sibling column are null).

![image](https://user-images.githubusercontent.com/14252330/40260601-e63b32c2-5ad1-11e8-87ef-72b591e762f4.png)

nmetulev commented:
Thanks @kbrons for researching this. Moving to 3.1 - let's revisit this in more details once we ship 3.0

kbrons commented:
How should we continue with this issue? Should I keep investigating a better solution to take care of the underlying issues?

jamers99 commented:
When will this be merged into master and released?  We're experiencing this issue in our app and would like it fixed!

kbrons commented:
@michael-hawker we should discuss this for the next release

Kyaa-dost commented:
I am still seeing a similar behavior and no change on my end. This is something I can talk to @michael-hawker and will test out more.

michael-hawker commented:
As part of the #3062 I'd like to refactor GridSplitter, so I'll try and re-map these scenarios on top of that while testing.

michael-hawker commented:
Going to close this and factor in these discussions during the refactor of GridSplitter.
